Sources/Confidence/RemoteResolveConfidenceClient.swift

```diff
@@ -33,10 +34,10 @@ class RemoteConfidenceResolveClient: ConfidenceResolveClient {
             apply: applyOnResolve,
             sdk: Sdk(id: metadata.name, version: metadata.version)
         )
-
+        let header = telemetry.getSnapshot()
         do {
             let result: HttpClientResult<ResolveFlagsResponse> =
-            try await self.httpClient.post(path: ":resolve", data: request)
+            try await self.httpClient.post(path: ":resolve", data: request, header: header)
```

Comment:
What does this `header` param in the post function accept?
IMO the header is pretty likely to be a key-value "tuple" that's gonna be appended to other headers.
If that's the case we probably want to have a common header key `X-CONFIDENCE-TELEMETRY` and pass the full telemetry data as the value.

Reply:
I am concerned about using the header in general, as it might be handled differently by various proxies and might have different and more unpredictable size limitations. We would also create new code paths in the backend to deal with data in the header (minor point, but still worth mentioning in the tradeoff discussions)